server/tserver/src/main/java/org/apache/accumulo/tserver/metrics/TabletServerMetrics.java

```diff
@@ -63,8 +81,55 @@ private long getTotalEntriesWritten() {
     return FileCompactor.getTotalEntriesWritten();
   }
 
+  public static class TableMetrics {
```

Comment:
I did some digging to see if Micrometer had any support for dynamic tags. I found https://github.com/micrometer-metrics/micrometer/pull/4097, which essentially allows you to create templates for Meters, then these get registered when you supply the tags.  I'm wondering if you had seen this, and if not, if it would change your implementation here. I'm thinking we could create the templates (MeterProvider in Micrometer) when the servers start up, then just apply and remove based on the table id tags.